File: 0301 - 0600/0468. Validate IP Address/fang.js
/**
 * @param {string} IP
 * @return {string}
 */
var validIPAddress = function (IP) {
    const IPv4 = /^(25[0-5]|2[0-4]\d|((1\d{2})|([1-9]?\d)))(\.(25[0-5]|2[0-4]\d|((1\d{2})|([1-9]?\d)))){3}$/
    const IPv6 = /^((\d|[a-f]){1,4})(\:(\d|[a-f]){1,4}){7}$/i;
    if (IPv4.test(IP))
        return "IPv4";
    if (IPv6.test(IP))
        return "IPv6";
    return "Neither";
};

File: 0301 - 0600/0468. Validate IP Address/fang.cpp
class Solution {
public:
    string validIPAddress(string IP) {
        if (regex_match(IP, regex("^(25[0-5]|2[0-4][0-9]|((1[0-9]{2})|([1-9]?[0-9])))(\.(25[0-5]|2[0-4][0-9]|((1[0-9]{2})|([1-9]?[0-9])))){3}$")))
            return string("IPv4");
        if (regex_match(IP, regex("^(([0-9]|[a-f]|[A-F]){1,4})(\:([0-9]|[a-f]|[A-F]){1,4}){7}$")))
            return string("IPv6");
        return string("Neither");
    }
};